<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->
<robot name="solidworks2ros_try15">

  <link name="dummy"> 
  </link>

  <joint name="dummy_joint" type="fixed">
     <parent link="dummy"/>
     <child link="base_link"/>
  </joint>

  <link name="base_link">
    <inertial>
      <origin
        xyz="7.6537E-18 1.5448E-17 0.17209"
        rpy="0 0 0" />
      <mass
        value="21.571" />
      <inertia
        ixx="2.2692"
        ixy="-3.3662E-17"
        ixz="4.5516E-17"
        iyy="2.2218"
        iyz="-3.4003E-16"
        izz="0.49397" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://solidworks2ros_try15/meshes/base_link.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://solidworks2ros_try15/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>

  <link name="link_1">
    <inertial>
      <origin
        xyz="-0.0017596 2.8589E-08 0.0017597"
        rpy="0 0 0" />
      <mass
        value="8.4177" />
      <inertia
        ixx="0.076973"
        ixy="-3.4016E-06"
        ixz="-0.0012255"
        iyy="0.11232"
        iyz="-3.3985E-06"
        izz="0.076973" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://solidworks2ros_try15/meshes/link_1.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://solidworks2ros_try15/meshes/link_1.STL" />
      </geometry>
    </collision>
  </link>

  <joint name="joint_1"
    type="revolute">
    <origin
      xyz="0 0 1.397"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="link_1" />
    <axis
      xyz="0 1 0" />
    <limit
      lower="0"
      upper="50"
      effort="100"
      velocity="0.1" />
  </joint>

  <link name="link_2">
    <inertial>
      <origin
        xyz="1.3878E-17 -3.0921E-17 -0.19049"
        rpy="0 0 0" />
      <mass
        value="0.42802" />
      <inertia
        ixx="0.0052275"
        ixy="1.0603E-08"
        ixz="1.7867E-18"
        iyy="0.0052275"
        iyz="-8.7102E-19"
        izz="9.7461E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://solidworks2ros_try15/meshes/link_2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://solidworks2ros_try15/meshes/link_2.STL" />
      </geometry>
    </collision>
  </link>

  <joint name="joint_2"
    type="prismatic">
    <origin
      xyz="0.1143 0 1.397"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="link_2" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="0"
      upper="2"
      effort="100"
      velocity="0.1" />
  </joint>

  <link name="link_3">
    <inertial>
      <origin
        xyz="1.3878E-17 -3.0921E-17 -0.19049"
        rpy="0 0 0" />
      <mass
        value="0.42802" />
      <inertia
        ixx="0.0052275"
        ixy="1.0603E-08"
        ixz="1.7867E-18"
        iyy="0.0052275"
        iyz="-8.7102E-19"
        izz="9.7461E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://solidworks2ros_try15/meshes/link_2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://solidworks2ros_try15/meshes/link_2.STL" />
      </geometry>
    </collision>
  </link>

  <joint name="joint_3"
    type="prismatic">
    <origin
      xyz="0 0 0"
      rpy="0 0 0" />
    <parent
      link="link_2" />
    <child
      link="link_3" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="0"
      upper="2"
      effort="100"
      velocity="0.1" />
  </joint>

    <link name="link_4">
    <inertial>
      <origin
        xyz="1.3878E-17 -3.0921E-17 -0.19049"
        rpy="0 0 0" />
      <mass
        value="0.42802" />
      <inertia
        ixx="0.0052275"
        ixy="1.0603E-08"
        ixz="1.7867E-18"
        iyy="0.0052275"
        iyz="-8.7102E-19"
        izz="9.7461E-05" />
    </inertial>
    <visual>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://solidworks2ros_try15/meshes/link_2.STL" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.79216 0.81961 0.93333 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://solidworks2ros_try15/meshes/link_2.STL" />
      </geometry>
    </collision>
  </link>

  <joint name="joint_4"
    type="prismatic">
    <origin
      xyz="0 0 0"
      rpy="0 0 0" />
    <parent
      link="link_3" />
    <child
      link="link_4" />
    <axis
      xyz="0 0 -1" />
    <limit
      lower="0"
      upper="2"
      effort="100"
      velocity="0.1" />
  </joint>

  <!-- Gazebo position controller -->
  <transmission name ="trans_1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_1">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_1">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name ="trans_2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_2">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_2">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name ="trans_3">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_3">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_3">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name ="trans_4">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="joint_4">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor_4">
        <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
        <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <!-- Gazebo plugin for controlling the rope -->
  <gazebo>
    <plugin name="ros_control" filename="libgazebo_ros_control.so">
    </plugin>
  </gazebo>

</robot>